test/suite/test_checkpoint23.py

```diff
@@ -0,0 +1,140 @@
+    # For now there is no point running it on VLCS (and it'll fail) because the checkpoint
+    # cleanup code does not handle VLCS pages. But the VLCS case should be enabled if that
+    # changes, e.g. as part of VLCS fast-delete support.
+    format_values = [
+        ('row_string', dict(key_format='S', value_format='S')),
+        #('column', dict(key_format='r', value_format='S')),
+    ]
+
+    scenarios = make_scenarios(format_values)
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def get_tree_stat(self, ds, stat):
+        stat_cursor = self.session.open_cursor('statistics:' + ds.uri, None, 'statistics=(all)')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def large_updates(self, ds, nrows, primary_value, secondary_value, commit_ts):
+        session = self.session
+        cursor = session.open_cursor(ds.uri)
+        for i in range(1, nrows + 1):
+            session.begin_transaction()
+            cursor[ds.key(i)] = secondary_value if i % 10 == 1 else primary_value
+            session.commit_transaction('commit_timestamp=' + self.timestamp_str(commit_ts))
+        cursor.close()
+
+    def large_removes(self, ds, nrows, commit_ts):
+        session = self.session
+        cursor = session.open_cursor(ds.uri)
+        for i in range(1, nrows + 1):
+            session.begin_transaction()
+            cursor.set_key(ds.key(i))
+            self.assertEqual(cursor.remove(), 0)
+            session.commit_transaction('commit_timestamp=' + self.timestamp_str(commit_ts))
+        cursor.close()
+
+    def test_checkpoint23(self):
+        uri = 'table:checkpoint23'
+        nrows = 10000
+
+        # Create a table.
+        ds = SimpleDataSet(self, uri, nrows,
+            key_format=self.key_format, value_format=self.value_format)
+        ds.populate()
+
+        # Pin oldest and stable timestamp to 1.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(1) +
+            ',stable_timestamp=' + self.timestamp_str(1))
+
+        # Use a seconary value large enough to be an overflow item.
+        value_a = "aaaaa" * 100
+        value_b = "bbbbb" * 10000
+
+        # Write a bunch of data at time 10.
+        self.large_updates(ds, nrows, value_a, value_b, 10)
+
+        # Delete it all at time 20.
+        self.large_removes(ds, nrows, 20)
+
+        # Mark it stable at 30 and checkpoint it.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(30))
+        self.session.checkpoint()
+
+        # Shut down and restart so nothing's in memory.
+        self.reopen_conn()
+
+        # Check how many overflow items we have.
+        initial_overflow_items = self.get_tree_stat(ds, stat.dsrc.btree_overflow)
+
+        # Shut down and restart again so nothing's in memory.
+        self.reopen_conn()
```

Comment:
I don't clearly understand the need for the second reopen_conn(), Is the recovery loads the overflow pages back into memory? Do we really need that?

Reply:
The stat pass will load the tree into memory to count the overflow items. (I believe. At least, the code in bt_stat.c that does the counting uses in-memory pages.)

Reply:
(I checked; it doesn't work without the second reopen. Added a comment in b27fd50b0.)

Reply:
Thanks for confirming it.

Reply:
YW :-)